pages/docs/couple-your-code/couple-your-code-just-in-time-mapping.md

```diff
@@ -0,0 +1,134 @@
+
+## Concept and API
+
+Using Just-in-time mapping requires smaller configuration changes in the precice configuration file and the source code of the adapter. The following configuration is an example for a just-in-time mapping preCICE configuration of our example participant called "SolverOne":
+
+```xml
+  <participant name="SolverOne">
+    <receive-mesh name="ReceivedMeshName" from="SolverTwo" api-access="true" />
+    <!-- data reading and writing is performed on MeshTwo, which is a received mesh with api-access enabled -->
+    <read-data name="Velocities" mesh="ReceivedMeshName" />
+    <write-data name="Forces" mesh="ReceivedMeshName" />
+    <!-- define the just-in-time mapping (read-consistent) to read the velocities, note the empty "to" mesh for the read direction-->
+    <mapping:nearest-neighbor direction="read" from="ReceivedMeshName" constraint="consistent" />
+    <!-- define the just-in-time mapping (write conservative) to write forces just-in-time, note the empty "from" mesh for the write direction-->
+    <mapping:nearest-neighbor direction="write" to="ReceivedMeshName" constraint="conservative" />
+  </participant>
+```
+
+Here, "SolverOne" now defines a just-in-time mapping access in read-direction for data "Velocities" and in write direction for data "Forces". In its core, using just-in-time mapping requires three changes in the configuration file:
+
+1. `api-access` needs to be enabled on the `receive-mesh` (similar to the direct access), "ReceivedMeshName" in our example above.
+2. The `read-data` and `write-data` tags need to use the received mesh with api-access enabled.
+3. The mapping only defines (depending on the direction) a "from" or a "to" mesh. For read-consistent mappings, only the "from" mesh must be defined. For write-conservative mappings, only the "to" mesh must be defined. In both cases, the defined mesh needs to match the received mesh with api-access enabled (which is also used in the `read-data` or `write-data` tag).
+
+At its core, the API now makes use of two new API functions, called `mapAndReadData` and `mapAndWriteData`. Both functions are very similar to the analogous functions `readData` and `writeData`: the only difference is that the functions take spatial coordinates as function arguments instead of static vertexIDs stemming from the preCICE initialization. In addition to `mapAndReadData` and `mapAndWriteData`, the access region needs to be defined by "SolverOne" using `setMeshAccessRegion`. The full code example reads the following:
```

Comment:
```suggestion
The API makes use of two new API functions, called `mapAndReadData` and `mapAndWriteData`. Both functions are very similar to the analogous functions `readData` and `writeData`: The only difference is that the functions take spatial coordinates as function arguments instead of static vertexIDs. In addition to `mapAndReadData` and `mapAndWriteData`, the access region needs to be defined by "SolverOne" using `setMeshAccessRegion`. A full code example reads as follows:
```